### PR TITLE
Added worker.py to cover worker integration tests

### PR DESCRIPTION
**Pull request by aaijazi**
@thompsonja - this is my attempt at making a more environment-unaware, readable integration test. 

@alainjobart 

I still want to add one more test for if a destination mysql is killed during worker copy.

### CODE REVIEW

File: test/worker.py [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+
+
+class TestSplitCloneResiliency(unittest.TestCase):
+  """Tests that the SplitClone worker is resilient to particular failures."""
+
+  def _init_keyspace(self):
+    """Creates a `test_keyspace` keyspace, and sets a sharding key."""
+    utils.run_vtctl(['CreateKeyspace', 'test_keyspace'])
+    utils.run_vtctl(['SetKeyspaceShardingInfo', 'test_keyspace',
+                     'keyspace_id', KEYSPACE_ID_TYPE])
```
Comment (alainjobart) [inline test/worker.py]: minor nitpick, CreateKeyspace takes -sharding_column_name and -sharding_column_type so you can combine these two.

**Comment by alainjobart** [comment]
LGTM

**Comment by aaijazi** [comment]
@thompsonja I don't actually know :). I think if you're polling for a "Done" variable, which is set immediately before shutting down a process, it could be short lived on the order of microseconds. I'm polling for the "cleaning up" state instead of the "done" state for that reason, among others.

**Comment by aaijazi** [comment]
@alainjobart @thompsonja added a tearDown and a MySQL test, PTAL

File: test/worker.py [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+        tablet.create_db('vt_test_keyspace')
+      tablet.start_vttablet(wait_for_state=None)
+
+    # Wait for tablet state to change after starting all tablets. This allows
+    # us to start all tablets at once, instead of sequentially waiting.
+    for tablet in shard_tablets.all_tablets:
+      tablet.wait_for_vttablet_state(wait_state)
+
+    # Reparent to choose an initial master
+    utils.run_vtctl(['ReparentShard', '-force', 'test_keyspace/%s' % shard_name,
```
Comment (alainjobart) [inline test/worker.py]: InitShardMaster there (without -force), ReparentShard is gone

**Comment by aaijazi** [comment]
Ended up removing a lot of the code that I added, since it's no longer needed with the improvements that have been happening in Vitess (particularly the improved reparent). The test should also be a lot more robust now.

PTAL, should be ready to merge. (although I'll squash all commits before merging)

**Comment by enisoc** [comment]
Could you add the new test to test/config.json too?

**Comment by alainjobart** [comment]
LGTM, this looks good now. Thanks for adding so much documentation in the test itself, it really helps.

Some of the classes you define (like the ShardTablets list) may be useful in other tests too...

**Comment by aaijazi** [comment]
@enisoc: done

@alainjobart: yep, I originally started out writing this test so that it could be a model test that we could base other tests off of. A lot of the reusable stuff that I wanted to write didn't pan out or became obsolete though, so I'm keeping stuff localized here until there's another use case.

**Pull request merged**